### PR TITLE
manish-updated-payment-history-API

### PR DESCRIPTION
**Pull request by kalchatelikhit**
Sir, Manish here, Please check it once...

### CODE REVIEW

File: src/main/java/com/srans/nestserver/controller/PaymentController.java [hunk 1]
```diff
@@ -39,56 +41,133 @@
...
+		logger.info("IN::GET::/payments::getPaymentHistory::" + userId + "::" + role);
+		List<PaymentHistory> paymentList = new ArrayList<>();
+
+		PaymentHistory payment = new PaymentHistory();
+		if (role != null && role.equalsIgnoreCase(NSConstants.ROLE_TENANT)) {
+
+			paymentRepository.getPaymentHistory(userId).stream().forEach(payments -> {
+
+				if (payments.getUserId() == userId) {
+					payment.setId(payments.getId());
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/PaymentController.java:73]: Why this if conditions again, u already queries with userId meaning only those the records matching corresponding userId will be returned from that caller, this condition not required u can remove that

File: src/main/java/com/srans/nestserver/controller/PaymentController.java [hunk 2]
```diff
@@ -39,56 +41,133 @@
...
+					payment.setPaymentThrough(payments.getPaymentThrough());
+					payment.setDepositAmount(payments.getDepositAmount());
+					payment.setDiscountAmount(payments.getDiscountAmount());
+					payment.setAmountToBePaid(payments.getAmountToBePaid());
+
+				}
+			});
+
+			List<Object[]> response = paymentRepository.getTenantInfo(userId);
+			// List<PaymentHistory> paymentHistory=new ArrayList<PaymentHistory>();
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/PaymentController.java:86]: This is good

File: src/main/java/com/srans/nestserver/controller/PaymentController.java [hunk 3]
```diff
@@ -39,56 +41,133 @@
...
+			}
+
+			Room room = roomRepository.getOne(roomId);
+			payment.setRoomName(room.getRoomName());
+			payment.setRoomType(room.getRoomType());
+			paymentList.add(payment);
+
+		} else {
+
+			List<PaymentHistory> ll = new ArrayList<>();
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/PaymentController.java:103]: Change variable name, don't l1

File: src/main/java/com/srans/nestserver/controller/PaymentController.java [hunk 4]
```diff
@@ -39,56 +41,133 @@
...
+
+				List<Object[]> response = paymentRepository.getTenantInfo(payments.getUserId());
+
+				Long roomId = 0L;
+
+				for (Object[] p : response) {
+
+					roomId = (Long) p[0];
+					System.out.println(p[1]);
+					paymentHistory.setName((String) p[1]);
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/PaymentController.java:123]: Remove System.out.print

File: src/main/java/com/srans/nestserver/controller/PaymentController.java [hunk 5]
```diff
@@ -39,56 +41,133 @@
...
+
+				}
+				Room room = roomRepository.getOne(roomId);
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/PaymentController.java:126]: Looks like some of and else as duplicate code, check it once, if that's ok then fine

**Review by srans-admin — APPROVED** [review]
Asked for few modifications,one implemented, just commit your changes to same PR and merge

**Comment by Manish-Srans** [comment]
Resolved conflicts

**Pull request merged**